src/rich-text/plugins/link-mod-click-handler.ts

```diff
@@ -0,0 +1,41 @@
+import { Plugin, EditorState } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+import { Mark } from "prosemirror-model";
+import { richTextSchema as schema } from "../schema";
+
+/**
+ * Returns the mark at cursor if it is of type `link`
+ * @param state The current editor state
+ */
+function findLinkAtCursor(state: EditorState): Mark {
```

Comment:
I think we can chuck this functionality into the `link-tooltip` plugin and avoid having to add another plugin. I'd wait on this for the time being though, as I'm nearly finished with a total overhaul of that file and changing it would cause some nasty merge conflicts.